challenge.rb

```diff
@@ -1,26 +1,34 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  temp = []
+  input.each do |element|
+  	temp.push( element.capitalize )
+  end
+  return temp
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  temp = [input.detect { |x| x == "dog"}]
+  return temp
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  temp = input.reject {|x| x=="dog"}
+  return temp
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  temp = input.first(2)
+  return temp
```

Comment:
Ruby implicitly returns the final value of a method, so Rubyists avoid explicit calls to `return` where possible.

Also, there's no need to assign the result to a local variable.

This example could be written:

```ruby
input.first(2)
```